internal/controller/serviceaccount_controller.go

```diff
@@ -171,17 +149,53 @@ func (r *serviceAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *serviceAccountReconciler) provisionSecretForAccount(
+	ctx context.Context, sa *corev1.ServiceAccount, principal string, accountIndex int, totalAccounts int,
+) (expiresAt time.Time, _ error) {
+	name := secretNameIndexed(sa, accountIndex)
+	logger := log.FromContext(ctx).WithValues("secret", name)
+	if totalAccounts > 1 {
+		logger = logger.WithValues("accountIndex", accountIndex)
+	}
```

Comment:
IMO this log value can be attached regardless of the total number of principals.